src/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -249,13 +252,19 @@ let exec_tap_ctl () =
   let pid_and_minor_to_sr_and_vdi = Utils.exec_cmd (module Process.D) ~cmdstring:tap_ctl ~f:process_line in
   let minor_to_sr_and_vdi = List.map snd pid_and_minor_to_sr_and_vdi in
   begin
-    if !vdi_to_vm_map = [] then begin
-      D.debug "VDI-to-VM map is empty; updating...";
-      update_vdi_to_vm_map ()
-    end else
       let disappeared_pids = Listext.List.set_difference !previous_map pid_and_minor_to_sr_and_vdi in
       let unmapped_vdi_pids = List.filter (fun (_, (_, (_, vdi))) -> not (List.mem_assoc vdi !vdi_to_vm_map)) pid_and_minor_to_sr_and_vdi in
 
+      let reason_for_updating_vdi_to_vm_map =
+        let newly_discovered = List.filter (fun (_, (_, (_, new_vdi))) -> List.for_all (fun (_, (_, (_, prev_vdi))) -> new_vdi <> prev_vdi) !previous_map) unmapped_vdi_pids in
+        if newly_discovered <> []
+        then
+          let pid_vdis_to_string pids = pids |> List.map (fun (pid, (_, (_, vdi))) -> Printf.sprintf "%d (%s)" pid vdi) |> String.concat "; " in
+          Some (Printf.sprintf "discovered new VDI(s): [%s]" (pid_vdis_to_string newly_discovered))
+        else if Unix.time () -. !last_time_vdi_to_vm_map_updated > 60. *. 5. (* 5 minutes *)
+        then Some "map was last updated over 5 minutes ago"
```

Comment:
I think we only need to do this when `unmapped_vdi_pids` is empty, otherwise we will update this map more often than before (i.e. every 5m, whereas before if everything was working well this list would've expected to get cleared completely, other than when bugs like XSI-400 are triggerred).

Reply:
Yes i agree with you here, but do you mean 'when `unmapped_vdi_pids` is non-empty'?

Reply:
Yes